include/veriblock/blockchain/blockchain_util.hpp

```diff
@@ -124,6 +123,9 @@ bool recoverEndorsements(ProtectedBlockTree& ed_,
 template <typename Block>
 void assertBlockCanBeRemoved(const Block& block);
 
+template <typename Block>
+void assertInsertBlockHeader(const Block& block);
```

Comment:
you can add default implementation and specify only VBK variant

Reply:
fixed